compiler-core/src/language_server/code_action.rs

```diff
@@ -1360,6 +1360,179 @@ impl<'a> AddAnnotations<'a> {
     }
 }
 
+/// Code action to add type annotations to all top level definitions
+///
+pub struct AnnotateTopLevelTypeDefinitions<'a> {
+    module: &'a Module,
+    params: &'a CodeActionParams,
+    edits: TextEdits<'a>,
+    printer: Printer<'a>,
+    is_hovering_definition: bool,
+}
+
+impl<'a> AnnotateTopLevelTypeDefinitions<'a> {
+    pub fn new(
+        module: &'a Module,
+        line_numbers: &'a LineNumbers,
+        params: &'a CodeActionParams,
+    ) -> Self {
+        Self {
+            module,
+            params,
+            edits: TextEdits::new(line_numbers),
+            // We need to use the same printer for all the edits because otherwise
+            // we could get duplicate type variable names.
+            printer: Printer::new_without_type_variables(&module.ast.names),
+            is_hovering_definition: false,
+        }
+    }
+
+    pub fn code_actions(mut self) -> Vec<CodeAction> {
+        self.visit_typed_module(&self.module.ast);
+
+        // We only want to trigger the action if we're over one of the definition in
+        // the module
+        if !self.is_hovering_definition {
+            return vec![];
+        };
+
+        let mut action = Vec::with_capacity(1);
+        CodeActionBuilder::new("Annotate all top level type definitions")
```

Comment:
I don't think the name is correct the code action is not annotating _type_ definitions, I'd call it "Annotate all top level definitions"